airbyte-integrations/connectors/source-github/source_github/source.py

```diff
@@ -114,7 +130,10 @@ def _get_branches_data(selected_branches: str, full_refresh_args: Dict[str, Any]
     def check_connection(self, logger: AirbyteLogger, config: Mapping[str, Any]) -> Tuple[bool, Any]:
         try:
             authenticator = self._get_authenticator(config)
-            repositories = self._generate_repositories(config=config, authenticator=authenticator)
+            # In case of getting repository list for given organization was
+            # successfull no need of checking stats for every repository within
+            # that organization.
+            repositories, _ = self._generate_repositories(config=config, authenticator=authenticator)
```

Comment:
- if the user provides only organization repos (repositories would be empty?), then the `check_connection` doesn't check anything?
- if repositories is a very long list of repo names, we still have the same issue?

Wouldn't it be better to limit the for loop at instead?
https://github.com/airbytehq/airbyte/blob/0eba52e002235caab14cdbb8f4c5847f8b8b0930/airbyte-integrations/connectors/source-github/source_github/source.py#L123

or at least limit to one public and one private from repositories, and maybe one for each organization?

Reply:
> if the user provides only organization repos (repositories would be empty?), then the check_connection doesn't check anything?

No, in this case it would try to get repo list from organization, pls look at unittest there is a case for that.

> if repositories is a very long list of repo names, we still have the same issue?

Yes, but this was an issue with organization with a lot of repos within. In case  if we get large list of repos it could be from different accounts and organizations and we should check all of them.

Reply:
> Wouldn't it be better to limit the for loop at instead? or at least limit to one public and one private from repositories, and maybe one for each organization?

In this case if user provide some private repo he has no access, check could falsely  success. In this commit I check every single repo and organization (without checking repo within org cause we "repo" scope should enable us to access every repo for that org). This fix making minimal sufficient requests to check that further read wont fail.

 As an optimization we could group single repo by organization but this defect is not about long list of repos, its about organization with a lot of repos.